### PR TITLE
(wixtoolset) Changed to AU package + Modernized

### PR DESCRIPTION
**Pull request by AdmiringWorm**
- Implemented AU for this package so we don't need to manually updated it each time.
- Embedded the software installer (not entirely sure if we also need to embedd the source, or if it's enough to just provide the link to the source repo)
- Updated the uninstall script so we don't need to change the registry key used for each update (tested with both 3.11 and 3.10)
- Updated the metadata (removed releaseNotes since I can't find the release notes anywhere anymore)

**NOTE: Description should be updated by someone more familiar with WiX than me**

@gep13 mind having a look since you're the original maintainer of this package.

### CODE REVIEW

**Comment by AdmiringWorm** [comment]
appveyor build is failing because I've forgotten to take into account installing pre-releases. Need to look into that.

File: automatic/wixtoolset/tools/chocolateyUninstall.ps1 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  $key | % {
+    $packageArgs['file'] = "$($_.BundleCachePath)"
+
+    Uninstall-ChocolateyPackage @packageArgs
+  }
+} elseif ($key.Count -eq 0) {
+  Write-Warning "$packageName has already been uninstalled by other means."
+} elseif ($key.Count -gt 1) {
+  Write-Warning "$($key.Count) matches found!"
+  Write-Warning "To prevent accidental data loss, no programs will be uninstall."
```
Comment (gep13) [inline automatic/wixtoolset/tools/chocolateyUninstall.ps1]: Minor nit-pick...

`no programs will be uninstalled.`

File: automatic/wixtoolset/tools/chocolateyUninstall.ps1 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  Write-Warning "Please alert the package maintainer the following keys were matched:"
```
Comment (gep13) [inline automatic/wixtoolset/tools/chocolateyUninstall.ps1]: Minor nit-pick...

`maintainer that the following keys were matched:`

File: automatic/wixtoolset/wixtoolset.nuspec [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <licenseUrl>http://wixtoolset.org/about/license/</licenseUrl>
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <projectSourceUrl>https://github.com/wixtoolset/wix3</projectSourceUrl>
+    <docsUrl>http://wixtoolset.org/documentation/</docsUrl>
+    <mailingListUrl>http://wixtoolset.org/documentation/</mailingListUrl>
+    <bugTrackerUrl>https://github.com/wixtoolset/issues/issues</bugTrackerUrl>
+    <tags>wixtoolset wix toolset msi xml admin</tags>
+    <summary>The most powerful set of tools available to create your Windows installation experience. Free and Open Source since 2004!</summary>
+    <description>
+The Windows Installer XML (WiX) is a toolset that builds Windows installation packages from XML source code. The toolset supports a command line environment that developers may integrate into their build processes to build MSI and MSM setup packages. The WiX toolset CodePlex project stores the stable releases and Mercurial repository.
```
Comment (gep13) [inline automatic/wixtoolset/wixtoolset.nuspec]: This seems fine to me.  The only comment would be that with CodePlex shutting down, going forward, the last statement will cease to be true.

Comment (AdmiringWorm) [inline automatic/wixtoolset/wixtoolset.nuspec]: Oh right, actually I didn't edit the description at all.

**Comment by AdmiringWorm** [comment]
damn, had to look hard for the typos you noticed. I couldn't see them at first.

File: automatic/wixtoolset/wixtoolset.nuspec (hunk 3 above)
Comment (AdmiringWorm) [inline automatic/wixtoolset/wixtoolset.nuspec]: btw, the description is imported from the Readme.md file when the package gets updated.

**Comment by AdmiringWorm** [comment]
@gep13 I fixed the typos, I also updated the description in the Readme.md file (and added a comment in the nuspec file to not edit the description there)

**Comment by AdmiringWorm** [comment]
@gep13 btw, what's the best way to update an existing pre-release package. How would we add the fix notation to that? just append `-20170421` and the end?

**Comment by gep13** [comment]
@AdmiringWorm hmm, not sure on that.  I "think" that might cause problems.

**Comment by AdmiringWorm** [comment]
@gep13 oh well, I'll merge it anyhow and see if I can't figure something out later.

**Pull request merged**